app/frontend/users/UserSessionResetPassword.res

```diff
@@ -5,7 +5,41 @@ let str = React.string
 
 let t = I18n.t(~scope="components.UserSessionResetPassword")
 
-let handleErrorCB = (setSaving, ()) => setSaving(_ => false)
+type state = {
+  authenticityToken: string,
+  token: string,
+  name: string,
+  email: string,
+  newPassword: string,
+  confirmPassword: string,
+  saving: bool,
+  newPasswordAnalysis: option<Zxcvbn.t>,
+}
+
+type action =
+  | UpdateNewPassword(string, string)
+  | UpdateConfirmPassword(string)
+  | UpdateSaving(bool)
+
+let reducer = (state, action) =>
+  switch action {
+  | UpdateNewPassword(newPassword, schoolName) => {
+      let newPasswordAnalysis = if Js.String2.length(newPassword) > 0 {
+        Zxcvbn.make(
+          ~password=newPassword,
+          ~userInputs=Js.Array2.concat(
+            Js.String2.split(state.name, " "),
+            Js.String2.split(state.email, "@"),
+          )->Js.Array2.concat([state.name, state.email, schoolName]),
+        )->Some
+      } else {
+        None
+      }
+      {...state, newPassword: newPassword, newPasswordAnalysis: newPasswordAnalysis}
```

Comment:
I have a feeling that this computation need to be done outside; Also do we need to cache `newPasswordAnalysis`, can't it be computed in render? 🤔

Reply:
No, we don't need to cache `newPasswordAnalysis. Let me try simplifying it!